python/pyarrow/tests/test_extension_type.py

```diff
@@ -267,14 +275,17 @@ def registered_period_type():
         pass
 
 
-def test_generic_ext_type():
-    period_type = PeriodType('D')
+@pytest.mark.parametrize("test_ext_class", [pa.ExtensionArray, PeriodArray])
+def test_generic_ext_type(test_ext_class):
+    period_type = PeriodType('D', test_ext_class)
     assert period_type.extension_name == "pandas.period"
     assert period_type.storage_type == pa.int64()
+    assert period_type.__arrow_ext_class__() == test_ext_class
 
 
-def test_generic_ext_type_ipc(registered_period_type):
-    period_type = PeriodType('D')
+@pytest.mark.parametrize("test_ext_class", [pa.ExtensionArray, PeriodArray])
+def test_generic_ext_type_ipc(registered_period_type, test_ext_class):
+    period_type = PeriodType('D', test_ext_class)
     storage = pa.array([1, 2, 3, 4], pa.int64())
     arr = pa.ExtensionArray.from_storage(period_type, storage)
```

Comment:
is `arr` already a PeriodArray? We should then also test that

Reply:
Indeed, I added a test checking this aspect